packages/gafl-webapp-service/src/pages/summary/contact-summary/__tests__/route.spec.js

```diff
@@ -64,318 +101,190 @@ const generateRequestMock = (permission, query, status = {}) => ({
+          })
+        )
         expect(summaryTable).toMatchSnapshot()
       })
     })
+  })
 
-    it('should have the newsletter is set have preferred method and if isLicenceForYou is true', async () => {
-      const permission = {
+  it.each([
+    ['Review or change the licence details', true, 'change_licence_details_you'],
+    ['Review or change the licence details other', false, 'change_licence_details_other']
+  ])('changeLicenceDetails should be %s when isLicenceForYou is %s', async (mssg, isLicenceForYou, mssgKey) => {
+    const permission = getMockPermission(
+      {},
+      {
         licenceLength: '1D',
-        licensee: {
-          ...addressAndContact,
-          postalFulfilment: false,
-          preferredMethodOfConfirmation: 'Prefer not to be contacted',
-          preferredMethodOfReminder: 'Prefer not to be contacted',
-          preferredMethodOfNewsletter: 'Email'
-        },
-        isLicenceForYou: true
-      }
-      const query = {
-        [HOW_CONTACTED.email]: 'Email'
+        isLicenceForYou
       }
-      const sampleRequest = generateRequestMock(permission, query)
-
-      const { summaryTable } = await getData(sampleRequest)
-
-      expect(summaryTable).toMatchSnapshot()
+    )
+    const mssgCatalog = getMockCatalog({
+      [mssgKey]: mssg
     })
-
-    it('should have the newsletter set to no if have preferred method and if isLicenceForYou is true', async () => {
-      const permission = {
-        licenceLength: '1D',
-        licensee: {
-          ...addressAndContact,
-          postalFulfilment: false,
-          preferredMethodOfConfirmation: 'Prefer not to be contacted',
-          preferredMethodOfReminder: 'Prefer not to be contacted',
-          preferredMethodOfNewsletter: 'Prefer not to be contacted'
-        },
-        isLicenceForYou: true
-      }
-      const query = {
-        [HOW_CONTACTED.email]: 'Prefer not to be contacted'
-      }
-      const sampleRequest = generateRequestMock(permission, query)
-
-      const { summaryTable } = await getData(sampleRequest)
-
-      expect(summaryTable).toMatchSnapshot()
+    const sampleRequest = getRequestMock({
+      permission,
+      catalog: mssgCatalog
     })
+    const { changeLicenceDetails } = await getData(sampleRequest)
+    expect(changeLicenceDetails).toBe(mssg)
+  })
 
-    it('should not have the newsletter row if isLicenceForYou is false', async () => {
-      const permission = {
-        licenceLength: '1D',
-        licensee: {
-          ...addressAndContact
-        },
-        isLicenceForYou: false
-      }
-      const sampleRequest = generateRequestMock(permission)
-
-      const { summaryTable } = await getData(sampleRequest)
-
-      expect(summaryTable).toMatchSnapshot()
+  it.each([
+    [HOW_CONTACTED.none, mssgs.no],
+    [null, mssgs.yes]
+  ])('returns yes or no to newsletter depending on user preference', async (preferredMethodOfNewsletter, expectedReturn) => {
```

Comment:
A small point, but rather than covering both cases in the test name, it should be tailored to provide a unique test name depending on the parameters that're being passed to the test. So we'd have 'newsletter is not sent when preferredMethodOfNewsletter is HOW_CONTACTED.none' and 'newsletter is sent when preferredMethodOfNewsletter is null'. 

You could do that by amending to this:
```
  it.each([
    { preferredMethodOfNewsletter: HOW_CONTACTED.none, expectedReturn: mssgs.no, action: 'not sent' },
    { preferredMethodOfNewsletter: null, expectedReturn: mssgs.yes, action: 'sent' }
  ])('newsletter is $action when preferredMethodOfNewletter is $preferredMethodOfNewsletter', async ({ preferredMethodOfNewsletter, expectedReturn }) => {
    const samplePermission = getMockPermission({
      preferredMethodOfNewsletter
    })
    const { summaryTable } = await getData(
      getRequestMock({
        permission: samplePermission
      })
    )
    expect(summaryTable[4].value.text).toBe(expectedReturn)
  })

```
Also, given that the second test description ends up as 'newsletter is sent when preferredMethodOfNewsletter is null', is that right? It doesn't sound right. I would have thought it should be sent when `preferredMethodOfNewsletter` is `HOW_CONTACTED.email`

Reply:
In general execution, `preferredMethodOfNewsletter` would never be null. So in place of `{ preferredMethodOfNewsletter: null, expectedReturn: mssgs.yes, action: 'sent' }`, do the following cases:
```
{ preferredMethodOfNewsletter: HOW_CONTACTED.email, expectedReturn: mssgs.yes, action: 'sent' }
{ preferredMethodOfNewsletter: HOW_CONTACTED.text, expectedReturn: mssgs.yes, action: 'sent' }
{ preferredMethodOfNewsletter: HOW_CONTACTED.post, expectedReturn: mssgs.yes, action: 'sent' }
```
Tests form 'use cases' for the code, and although it's possible to pass 'null' for `preferredMethodOfNewsletter`, it wouldn't ever be set to null. It's more helpful to set actual, expected values that show the use of the code in expected operation.

Reply:
that makes alot more sense thanks!